openpype/hosts/blender/api/utils.py

```diff
@@ -343,27 +343,47 @@ def make_paths_absolute(source_filepath: Path = None):
         source_filepath (Path, optional): Filepath to remap paths from,
             in case file copy has been executed without paths remapping.
             Defaults to None.
+
+    Returns:
+        Set[bpy.types.ID]: Remapped datablocks.
     """
-    # In case no source filepath try naive system
-    if not source_filepath:
-        bpy.ops.file.make_paths_absolute()
-        return
 
-    for datablock in list(bpy.data.libraries) + list(bpy.data.images):
-        try:
-            if datablock and datablock.filepath.startswith("//"):
+    relative_datablocks = set()
+    for data_name in dir(bpy.data):
+        data_collection = getattr(bpy.data, data_name)
+        if not isinstance(data_collection, bpy.types.bpy_prop_collection):
+            continue
+        for datablock in data_collection.values():
+            if not hasattr(datablock, "filepath"):
+                break
+            if (
+                not datablock.is_property_readonly("filepath")
+                and isinstance(datablock.filepath, str)
+                and datablock.filepath.startswith("//")
+            ):
+                relative_datablocks.add((datablock, datablock.filepath))
+
+    if source_filepath:
+        for datablock, filepath in relative_datablocks:
+            try:
                 datablock.filepath = str(
                     Path(
                         bpy.path.abspath(
-                            datablock.filepath,
+                            filepath,
                             start=source_filepath.parent,
                         )
                     ).resolve()
                 )
-        except (RuntimeError, ReferenceError, OSError) as e:
-            print(e)
+            except (RuntimeError, ReferenceError, OSError) as err:
+                print(err)
+    else:
+        bpy.ops.file.make_paths_absolute()
```

Comment:
If always executed (no `break` in loop) it is recommended to keep it out of the `else`.